### DISCUSSION TITLE
Solution for garage door

### DISCUSSION THREAD

**Discussion by ikino** [arendst/Tasmota#21138] [Q&A]
Hello

How I can do garage door controller? I have esp with tasmota, GPIO3 is Relay_i, GPIO4 is Switch_n. On main page there are toogle button and state of relay.

How I can change state of switch instead of relay? How I can toogle caption of button depending on switch state?

Thanks...

![image](https://github.com/arendst/Tasmota/assets/43843741/786fcd65-48cd-4ff5-ae61-d4df4c395262)

**Comment by sfromis** [comment]
A `Switch` is an *input* pin, you can only change it by operating the electrical input. By default switch actions will trigger Relay actions, which is then reflected on the front page. The *input* is not reflected.

More generally, the Tasmota web UI is **not** meant as a "dashboard" for daily operation, having a "full dashboard" would be beyond the capacity of ESP8266 when Tasmota supports a large number of features.

**Comment by ivanvach** [comment]
As they say in Military KISS (keep it simple ...)
If your garage door is operated by a simple push button then keep it and use it.
Connect your Tasmota device's relay in parallel to the push button to operate it remotely from your home automation.
That's all. You don't need "switch" input to your Tasmota.
If your garage door unit is one of the newer "smart" units than there is encoded protocol between the wall button (that is "smart" ) and the motor. Things are getting much more complicated. Look online for ratgdo .
